src/game_updater/GameUpdater.cpp

```diff
@@ -100,69 +136,76 @@ void GameUpdater::UpdatePlayer(std::shared_ptr<GameState> game_state) {
+            std::dynamic_pointer_cast<MovingObject>(player->GetGround())) {
+      // Move player with moving object
+      glm::vec3 movementVector = moving->GetMovementVector();
+      player->SetVerticalVelocity(movementVector.y * moving->GetVelocity().y);
+      player->SetZVelocity(movementVector.z * moving->GetVelocity().z);
+      collision_width += movementVector.y * moving->GetVelocity().y;
+    } else {
+      // player is stuck to ground, make sure velocity is neutralized
+      player->SetVerticalVelocity(0.0f);
+    }
   } else {
     // player is in the air, apply gravity
     player->SetVerticalVelocity(previous_player_velocity - PLAYER_GRAVITY);
   }
-  player->SetPosition(
-      previous_player_position +
-      glm::vec3(DELTA_X_PER_TICK, player->GetVerticalVelocity(), 0));
-  AxisAlignedBox future_player_box = player->GetBoundingBox();
-
-  std::vector<std::shared_ptr<Platform>> colliding_platforms =
-      GetCollidingObjects(future_player_box,
-                          game_state->GetLevel()->getPlatforms());
-
-  std::vector<std::shared_ptr<Note>> colliding_collectibles =
-      GetCollidingObjects(future_player_box,
-                          game_state->GetLevel()->getNotes());
-
-  for (int i = 0; i < colliding_collectibles.size(); i++) {
-    if (!colliding_collectibles[i]->GetCollected()) {
-      colliding_collectibles[i]->SetCollected();
-      game_state->GetPlayer()->SetScore(game_state->GetPlayer()->GetScore() +
-                                        1);
+  player->SetPosition(player->GetPosition() +
+                      glm::vec3(DELTA_X_PER_TICK, player->GetVerticalVelocity(),
+                                player->GetZVelocity()));
+
+  // collide!
+  std::shared_ptr<std::unordered_set<std::shared_ptr<GameObject>>>
+      collidingObjects = GetCollidingObjects(player->GetBoundingBox(),
+                                             game_state->GetLevel()->getTree());
+  std::vector<std::shared_ptr<GameObject>> colliding_platforms;
+  for (std::shared_ptr<GameObject> g : *collidingObjects) {
+    if (g->GetType() == ObjectType::COLLECTIBLE) {
+      if (std::shared_ptr<Collectible> c =
+              std::static_pointer_cast<Collectible>(g)) {
+        if (!c->GetCollected()) {
+          c->SetCollected();
+          game_state->GetPlayer()->SetScore(
+              game_state->GetPlayer()->GetScore() + 1);
+        }
+      }
+    } else if (g->GetType() == ObjectType::OBSTACLE) {
+      colliding_platforms.push_back(g);
     }
   }
 
-  if (colliding_platforms.size() > 1) {
-    // TODO(jarhar): further consider what is happening when we are colliding
-    // with two objects
-    Reset(game_state);
-  } else if (colliding_platforms.size() == 0) {
-    // No collision
-  } else {
-    // Handle the collision
-    std::shared_ptr<GameObject> colliding_object = colliding_platforms[0];
+  // Handle the collisions
+  for (std::shared_ptr<GameObject> colliding_object : colliding_platforms) {
```

Comment:
Does looping through colliding platforms like this seem to work well? I would think that it would be weird if you get snapped to two different ground platforms based on which comes last in this list, but if it works then I'm ok with it